src/control/cmd/daos_agent/mgmt_rpc.go

```diff
@@ -170,10 +175,24 @@ func (mod *mgmtModule) handleGetAttachInfo(reqb []byte, pid int32) ([]byte, erro
 		return nil, err
 	}
 
-	return mod.aiCache.getResponse(numaNode)
+	cacheResp, err := mod.aiCache.getResponse(numaNode)
+	if err != nil {
+		return nil, err
+	}
+
+	req := NewProcMonRequest(pid, drpc.MethodGetAttachInfo.ID())
+	mod.monitor.request <- req
```

Comment:
+1 on this idea, much cleaner.

Reply:
Is it legal to write 'mod.monitor.request <- NewProcMonRequest(pid, drpc.MethodGetAttachInfo.ID())' instead of making a convenience function?  The convenience function is fine with me if the one line approach isn't valid.

Reply:
I agree. I originally didn't have it as a helper function because it was only once instance but then I found the issue with not being able to register the process if the agent had the information cached. I made it into a function on the procMon struct now so it should be easier and hide the information.

@jbrosenz I tried what you suggested there and it had issues with the syntax and wouldn't allow me to do that. Its all hidden now so it should be better.

Reply:
Fixed this by adding the SubmitRequest function that Kris suggested.